index.d.ts

```diff
@@ -9,6 +9,8 @@ declare namespace findUp {
 	}
 }
 
+type Match = string | boolean | null | undefined;
```

Comment:
Should I put this on the namespace?

Reply:
I think we should type this one as:

```ts
type Match = string | typeof findUp.stop | undefined;
```

While JS is loose, we should try to keep the TS types strict.

---

> Also, should I put this type on the namespace?

Yes

Reply:
I agree in general, but in this case, if we are very strict, then people can't necessarily do:

```js
findUp((directory) => {
    return foo && bar;
});
```

... since `foo` might be `false` or `null`, etc. Is it worth losing that nice syntax?

Reply:
TS users will definitely say yes. That's why they're using TS; they want strict types. It doesn't affect the majority that is using plain JS.

TS users can do:

```
findUp((directory) => {
    return foo ? bar : undefined;
});
```

It's just a tiny bit more verbose.